### PR TITLE
migrate dashboard stored default filters to time range format

### LINKED ISSUE #5553: Time filter from older versions break exploring slices from dashboard
Upgrading from 0.26.3 to the latest master branch, using a dashboard with "from" and "to" filters stored as dashboard metadata in the previous version, like

` "default_filters": "{\"1234\": {\"__from\": \"15 days ago\", \"__to\": \"1 days ago\"}}"`

Following #4981 these are not converted to the new `__time_range` filter and not being treated as date options in `utils.merge_extra_filters` they are used as filters instead:

`"{"expressionType":"SIMPLE","subject":"__from","operator":"in","comparator":"15 days ago","clause":"WHERE","sqlExpression":null,"fromFormData":true,"filterOptionName":"63ad1398-eec0-449f-b0ae-7b8ab2c77b21"}"`

Since the operator is "in" and the comparator is not an array, it also breaks when exploring any slice from that dashboard:

`webpack:///./src/explore/AdhocFilter.js?:48 Uncaught TypeError: adhocMetric.comparator.join is not a function`

### PR DESCRIPTION
**Pull request by octaviancorlade**
Since #4981 `__from` and `__to` default filters from dashboards were treated as normal WHERE filters instead of being used to fill the time range options.

An alternative to this would be some legacy function to handle the previous format.

Fixes #5553

### CODE REVIEW

**Comment by mistercrunch** [comment]
I think @john-bodley worked on backend legacy filter handling recently here https://github.com/apache/incubator-superset/pull/5525

File: superset/migrations/versions/388242a6da7e_dash_json_add_time_range.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        for filter_id in default_filters:
+            filter = default_filters[filter_id]
+            if '__from' in filter or '__to' in filter:
+                time_range = '{} : {}'.format(
+                    filter.get('__from', ''), filter.get('__to', ''))
+                default_filters[filter_id] = {'__time_range': time_range}
+
+        metadata['default_filters'] = json.dumps(default_filters, indent=2)
+        dashboard.json_metadata = json.dumps(metadata, indent=2)
+        session.merge(dashboard)
```
Comment (john-bodley) [inline superset/migrations/versions/388242a6da7e_dash_json_add_time_range.py]: You shouldn’t need to explicitly merge the updated dashboard object as SQLAlechemy is aware of these mutation.

File: superset/migrations/versions/388242a6da7e_dash_json_add_time_range.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        session.commit()
```
Comment (john-bodley) [inline superset/migrations/versions/388242a6da7e_dash_json_add_time_range.py]: The commit should be outside of the for lool.

**Comment by octaviancorlade** [comment]
Updated

**Pull request closed without merging**